pkg/scheduler/scheduler.go

```diff
@@ -462,20 +480,91 @@ func (s *Scheduler) getAssignments(log logr.Logger, wl *workload.Info, snap *cac
 	return assignment, targets
 }
 
+// preemptableWorkloadSlice determines whether the given workload is eligible for
+// slice-based preemption under the ElasticJobsViaWorkloadSlices feature.
+//
+// If the feature gate `ElasticJobsViaWorkloadSlices` is not enabled, it returns nil.
+// Otherwise, it attempts to identify a preemptible workload slice target from the
+// provided snapshot that can be reclaimed to admit the given workload.
+//
+// Returns a list containing a single preemption target and the associated workload info
+// if a suitable preemptible slice is found, or nil if no such target exists.
+func preemptableWorkloadSlice(wl *workload.Info, snap *cache.Snapshot) ([]*preemption.Target, *workload.Info) {
```

Comment:
please move the new functions to `pkg/scheduler/workloadslicing.go`, just to offload the main scheduler file.

Reply:
done.